backport: fix warnings

**Is your feature request related to a problem? Please describe.**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
commit id:  00dac9644d194d8c30aee904130a8205f13f2d1f

check this pr and keep consistent with 8.0.

**Describe the solution you'd like**
<!-- A clear and concise description of what you want to happen. -->

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->

**Additional context**
<!-- Add any other context or screenshots about the feature request here. -->

There are just 2 warnings when building stonedb8.0, and same kind 
use `static_cast` to transform the type of `rcTables.size()` from `long unsigned int` to `int`
```
/home/stonedb/storage/tianmu/core/parameterized_filter.cpp: In member function ‘void Tianmu::core::ParameterizedFilter::UpdateMultiIndex(bool, int64_t)’:
/home/stonedb/storage/tianmu/core/parameterized_filter.cpp:1024:41: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Tianmu::core::JustATable*>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
 1024 |     for (int tableIndex = 0; tableIndex < rcTables.size(); ++tableIndex) {
      |                              ~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/home/stonedb/storage/tianmu/core/parameterized_filter.cpp: In member function ‘void Tianmu::core::ParameterizedFilter::FilterDeletedForSelectAll()’:
/home/stonedb/storage/tianmu/core/parameterized_filter.cpp:1562:41: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Tianmu::core::JustATable*>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
 1562 |     for (int tableIndex = 0; tableIndex < rcTables.size(); ++tableIndex) {
      |                              ~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

There are 3 warnings when build type is release, above is debug type
```bash
/home/stonedb/storage/tianmu/core/column_share.cpp: In member function ‘int Tianmu::core::ColumnShare::alloc_dpn(Tianmu::common::TX_ID, const Tianmu::core::DPN*)’:
/home/stonedb/storage/tianmu/core/column_share.cpp:239:12: warning: ignoring return value of ‘int ftruncate(int, __off64_t)’, declared with attribute warn_unused_result [-Wunused-result]
  239 |   ftruncate(dn_fd, capacity * sizeof(DPN));
      |   ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/stonedb/storage/tianmu/system/channel.cpp: In member function ‘Tianmu::system::Channel& Tianmu::system::Channel::lock(uint)’:
/home/stonedb/storage/tianmu/system/channel.cpp:51:50: warning: ‘%02d’ directive writing between 2 and 11 bytes into a region of size between 0 and 17 [-Wformat-overflow=]
   51 |     std::sprintf(sdatetime, "[%4d-%02d-%02d %02d:%02d:%02d.%06d]", cdt->tm_year + 1900, cdt->tm_mon + 1, cdt->tm_mday,
      |                                                  ^~~~
/home/stonedb/storage/tianmu/system/channel.cpp:51:29: note: using the range [-2147483648, 2147483647] for directive argument
   51 |     std::sprintf(sdatetime, "[%4d-%02d-%02d %02d:%02d:%02d.%06d]", cdt->tm_year + 1900, cdt->tm_mon + 1, cdt->tm_mday,
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:867,
                 from /usr/include/c++/9/cstdio:42,
                 from /usr/include/c++/9/ext/string_conversions.h:43,
                 from /usr/include/c++/9/bits/basic_string.h:6496,
                 from /usr/include/c++/9/string:55,
                 from /usr/include/c++/9/stdexcept:39,
                 from /usr/include/c++/9/array:39,
                 from /usr/include/c++/9/tuple:39,
                 from /usr/include/c++/9/mutex:38,
                 from /home/stonedb/storage/tianmu/system/channel.cpp:20:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 29 and 86 bytes into a destination of size 32
   36 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   37 |       __bos (__s), __fmt, __va_arg_pack ());
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/stonedb/storage/tianmu/types/rc_num.cpp: In member function ‘virtual Tianmu::types::BString Tianmu::types::RCNum::ToBString() const’:
/home/stonedb/storage/tianmu/types/rc_num.cpp:249:11: warning: ignoring return value of ‘char* gcvt(double, int, char*)’, declared with attribute warn_unused_result [-Wunused-result]
  249 |       gcvt(*reinterpret_cast<double *>(const_cast<int64_t *>(&value_)), 15, buf);
      |       ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

--- a/storage/tianmu/core/column_share.cpp
+++ b/storage/tianmu/core/column_share.cpp
@@ -236,7 +236,7 @@ int ColumnShare::alloc_dpn(common::TX_ID xid, const DPN *from) {
   // It is not portable to enlarge the file size after mmapping, but it seems to
   // work well on Linux. Otherwise we'll need to remmap the file, which in turn
   // requires locking because other threads may read simultaneously.
-  ftruncate(dn_fd, capacity * sizeof(DPN));
+  [[maybe_unused]] int tmp = ftruncate(dn_fd, capacity * sizeof(DPN));
   init_dpn(start[capacity - 1], xid, from);
   return capacity - 1;
 }

--- a/storage/tianmu/system/channel.cpp
+++ b/storage/tianmu/system/channel.cpp
@@ -45,7 +45,7 @@ Channel &Channel::lock(uint optional_sess_id) {
   if (time_stamp_at_lock_ && enabled_) {
     time_t curtime = time(nullptr);
     struct tm *cdt = localtime(&curtime);
-    char sdatetime[32] = "";
+    char sdatetime[64] = "";
     struct timespec time;
     clock_gettime(CLOCK_REALTIME, &time);
     std::sprintf(sdatetime, "[%4d-%02d-%02d %02d:%02d:%02d.%06d]", cdt->tm_year + 1900, cdt->tm_mon + 1, cdt->tm_mday,

--- a/storage/tianmu/types/rc_num.cpp
+++ b/storage/tianmu/types/rc_num.cpp
@@ -246,7 +246,7 @@ BString RCNum::ToBString() const {
     static int const SIZE(24);
     char buf[SIZE];
     if (core::ATI::IsRealType(attr_type_)) {
-      gcvt(*reinterpret_cast<double *>(const_cast<int64_t *>(&value_)), 15, buf);
+      [[maybe_unused]] char *tmp = gcvt(*reinterpret_cast<double *>(const_cast<int64_t *>(&value_)), 15, buf);
       size_t s = std::strlen(buf);
       if (s && buf[s - 1] == '.')
         buf[s - 1] = 0;

